hack/test/cert-e2e.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e
+
+# patch the certificates to renew every minute
+# (if set to 2 minutes, then the big timeout wait is about 5 minutes)
+kubectl patch certificate.cert-manager.io -n cert-manager olmv1-ca --type='merge' -p '{"spec":{"duration":"1h", "renewBefore":"59m"}}'
+kubectl patch certificate.cert-manager.io -n olmv1-system olmv1-cert --type='merge' -p '{"spec":{"duration":"1h", "renewBefore":"59m"}}'
+
+# delete the old secrets, so new secrets are generated
+kubectl delete secret -n cert-manager olmv1-ca
+kubectl delete secret -n olmv1-system olmv1-cert
+
+# delete the existing operator-controller, to force it to get a new secret
+# (deleting the secret itself isn't enough)
+kubectl delete pod -n olmv1-system -l control-plane=controller-manager
+kubectl wait --for=condition=Available --namespace=olmv1-system "deployment/operator-controller-controller-manager" --timeout="60s"
+
+# and then search through the previous logs
+function check_logs() {
+    kubectl logs -c manager -n olmv1-system -l control-plane=controller-manager | grep "updating certificate pool" >& /dev/null
```

Comment:
You're correct in that the cert-e2e only test the reloading of certificates. It would be _much_ more complex and time-consuming to test an end-to-end connection with catalogd. The renewal period would have to be adjusted to make this test possibly run 10+ minutes, which is undesirable. Now that there's a separate CertPoolWatcher, I can likely do it as a unit test; the code before was not amenable to a unit-test.

Reply:
I removed the e2e, and now use a unit test